src/services/ui/src/features/package-actions/legacy-shared-components.tsx

```diff
@@ -83,6 +86,14 @@ export const useDisplaySubmissionAlert = (header: string, body: string) => {
       if (location.pathname?.endsWith("/update-id")) {
         alert.setBannerDisplayOn("/dashboard");
         navigate("/dashboard");
+      } else if (data.isTe) {
+        if (isNewSubmission() || location.state?.from?.includes("dashboard")) {
+          alert.setBannerDisplayOn("/dashboard");
+          navigate(urlEmbedQuery("/dashboard", { tab: "waivers" }));
+        } else {
+          alert.setBannerDisplayOn("/details");
+          navigate(location?.state?.from ?? "/dashboard");
+        }
```

Comment:
non blocking since this is a legacy file that'll be folded into a new pattern, but stuff like nested if/else-if/if/else's get...gross. try keeping things like this a little less verbose while still clearly stating your intentions.

also, observing the larger goal here, was there any reason you didn't consider breaking out this logic and just returning what you need? hooks can call hooks, so you could have made a hook to do a lot of this math, and just called `setBannerDisplayOn` and `navigate` once without the if/else nesting

Reply:
Agreed. I just didn't put too much refactoring effort into it since it's scope is limited to one - two places and we will be getting rid of it with future refactors. My hope is that once we get TE's to use the same language as your current design all this code will be OBE